### PR TITLE
Add support for FOR UPDATE NOWAIT, SKIP LOCKED, OF

### PR DESCRIPTION
**Pull request by squareapartments**
Hi, due to inactivity of original PR #565 I've imported current changes and applied fixes as suggested by @x8lucas8x in their last comment. Please let me know if this is what you expected and/or if any other changes should be applied.

My changes move the initial implementation to MySQL and PostgreSQL dialects, as these databases seem to support these enhancements to FOR UPDATE. Tests are updated accordingly.

### CODE REVIEW

File: pypika/tests/test_selects.py [hunk 1]
```diff
@@ -378,6 +379,141 @@ def test_where_field_equals_for_update(self):
...
+                .select("*")
+                .where(self.t.foo == self.t.bar)
+                .for_update(nowait=True, skip_locked=True, of=("abc",))
+            )
+            quote_char = query_cls._builder().QUOTE_CHAR if isinstance(query_cls._builder().QUOTE_CHAR, str) else '"'
+            self.assertEqual(
+                'SELECT * '
+                'FROM {quote_char}abc{quote_char} '
+                'WHERE {quote_char}foo{quote_char}={quote_char}bar{quote_char} '
+                'FOR UPDATE OF {quote_char}abc{quote_char} NOWAIT'.format(
```
Comment (x8lucas8x) [inline pypika/tests/test_selects.py:489]: So SKIP LOCKED shouldn't be here?

**Comment by x8lucas8x** [comment]
Thanks. Overall I think that looks good. I've left a question though.

File: pypika/tests/test_selects.py (hunk 1 above)
Comment (squareapartments) [reply pypika/tests/test_selects.py:489]: The name suggests it should be both, but query synopsis for both PostgreSQL and MySQL says that `NOWAIT` and `SKIP LOCKED` are mutually exclusive.
PostgreSQL:
```
[ FOR { UPDATE | NO KEY UPDATE | SHARE | KEY SHARE } [ OF table_name [, ...] ] [ NOWAIT | SKIP LOCKED ] [...] ]
```
MySQL:
```
[FOR {UPDATE | SHARE}
        [OF tbl_name [, tbl_name] ...]
        [NOWAIT | SKIP LOCKED]
      | LOCK IN SHARE MODE]
```

So I split the test into two - testing both cases.

Comment (x8lucas8x) [reply pypika/tests/test_selects.py:489]: I see. Thanks.

**Comment by x8lucas8x** [comment]
@squareapartments can you please fix the black formatting issues? I will merge it then.

**Comment by squareapartments** [comment]
@x8lucas8x sure no problem. Do you also want me to fix the coverage issue?

**Comment by x8lucas8x** [comment]
@squareapartments no need. That's a marginal variation and you have provided tests.

**Comment by squareapartments** [comment]
Thanks! How often do you release pypika?

**Pull request merged**